app/src/main/kotlin/com/yubico/yubioath/ui/main/CredentialFragment.kt

```diff
@@ -214,14 +213,19 @@ class CredentialFragment : ListFragment(), CoroutineScope {
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
 
-        fun handleUrlCredential(context: Context, data: String) {
-            try {
-                val uri = Uri.parse(data)
-                CredentialData.fromUri(uri)
-                startActivityForResult(Intent(Intent.ACTION_VIEW, uri, context, AddCredentialActivity::class.java), REQUEST_ADD_CREDENTIAL)
-            } catch (e: IllegalArgumentException) {
+        fun handleUrlCredential(context: Context, data: String?) {
+            if (data != null) {
+                try {
+                    val uri = Uri.parse(data)
+                    CredentialData.fromUri(uri)
+                    startActivityForResult(Intent(Intent.ACTION_VIEW, uri, context, AddCredentialActivity::class.java), REQUEST_ADD_CREDENTIAL)
+                } catch (e: IllegalArgumentException) {
```

Comment:
can you add comment also which method throws that exception?...
if we can prevent it, It's better to check argument rather than throw exception...